src/components/TodoHeader.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { RefObject, useContext, useState } from 'react';
+import * as postServise from '../api/todos';
+import { TodosContext } from './TodoProvider';
+import { Errors } from '../types/Errors';
+
+type Props = {
+  handleChangeInput: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  title: string,
+  inputRef: RefObject<HTMLInputElement>;
+  setTitle: (valu: string) => void,
+};
+
+export const TodoHeader: React.FC<Props> = ({
+  handleChangeInput,
+  title,
+  inputRef,
+  setTitle,
+}) => {
+  const [isSubmiting, setIsSubmiting] = useState(false);
+  const {
+    todos,
+    setTodos,
+    setError,
+    USER_ID,
+  } = useContext(TodosContext);
+
+  const handleFormSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!title.trim()) {
+      setError(Errors.ERRORS_EMPTY_TITLE);
+
+      return;
+    }
+
+    setIsSubmiting(true);
+
+    try {
+      const newTodo = await postServise.createTodo({
+        completed: false,
+        title: title.trim(),
+        userId: USER_ID,
+      });
+
+      setTodos(currentTodo => [...currentTodo, newTodo]);
+      setTitle('');
+    } catch (newError) {
+      setError(Errors.UNABLE_ADD);
+      throw newError;
```

Comment:
do not throw an error here - you handled it already, so no need to throw it again. It will lead to a page crash (because you don't try to catch the error thrown here somewhere else)